fmralign/alignment_methods.py

```diff
@@ -731,3 +739,256 @@ def transform(self, X, verbose=False):
         )
 
         return np.array(reconstructed_signal, dtype=np.float32)
+
+
+class FugwAlignment:
+    """Wrapper for FUGW alignment"""
+
+    def __init__(
+        self,
+        alpha_coarse=0.5,
+        alpha_fine=0.5,
+        rho_coarse=1.0,
+        rho_fine=1.0,
+        eps_coarse=1.0,
+        eps_fine=1.0,
+        radius=5,
+        anisotropy=(3, 3, 3),
+        reg_mode="independent",
+        divergence="kl",
+    ) -> None:
+        """Initialize FUGW alignment
+
+        Parameters
+        ----------
+        n_samples : int, optional
+            Number of samples from the embedding, by default 300.
+        alpha_coarse : float, optional, by default 0.5.
+        rho_coarse : float, optional, by default 1.
+        eps_coarse : float, optional, by default 1.
+        alpha_fine : float, optional, by default 0.5.
+        rho_fine : float, optional, by default 1.
+        eps_fine : float, optional, by default 1e-6.
+        radius : int, optional
+            Radius around the sampled points in mm, by default 5.
+        anisotropy : tuple, optional.
+            Anisotropy of the fmri mask, by default (3, 3, 3)
+        reg_mode : str, optional
+            Regularization mode, by default "independent"
+        divergence : str, optional
+            Divergence used in the FUGW alignment, by default "kl".
+        """
+        self.alpha_coarse = alpha_coarse
+        self.rho_coarse = rho_coarse
+        self.eps_coarse = eps_coarse
+        self.alpha_fine = alpha_fine
+        self.rho_fine = rho_fine
+        self.eps_fine = eps_fine
+        self.radius = radius
+        self.anisotropy = anisotropy
+        self.reg_mode = reg_mode
+        self.divergence = divergence
+
+    def _normalize(self, X):
+        """Normalize the input data"""
+        return (X / np.linalg.norm(X, axis=1).reshape(-1, 1)).T
+
+    def _prepare_geometry_embedding(self, segmentation, n_landmarks, verbose):
+        """Compute the normalized geometry embedding"""
+        geometry_embedding = lmds.compute_lmds_volume(
+            segmentation,
+            k=12,
+            n_landmarks=n_landmarks,
+            anisotropy=self.anisotropy,
+            verbose=verbose,
+        ).nan_to_num()
+
+        (
+            geometry_embedding_normalized,
+            max_distance,
+        ) = coarse_to_fine.random_normalizing(geometry_embedding)
+
+        return (
+            geometry_embedding,
+            geometry_embedding_normalized,
+            max_distance,
+        )
+
+    def _sample_geometry(self, segmentation, geometry_embedding, n_samples):
+        """Sample the geometry of the mask"""
+        return coarse_to_fine.sample_volume_uniformly(
+            segmentation,
+            embeddings=geometry_embedding,
+            n_samples=n_samples,
+        )
+
+    def fit(
+        self,
+        X,
+        Y,
+        segmentation,
```

Comment:
sklearn types estimators tend to mvoe all parameters  (beyond X,Y) to the builder and have a very homgeneous fit(X,Y) API. Could we at least get closer to it ?

Reply:
We should have a discussion on the API...